routes/api.js

```diff
@@ -1,29 +1,117 @@
+			//check if user accesing machine exist
+		 	!user ? res.sendStatus(403) : console.log(user);
+			if (user.dataValues.status === 'Admin') {
+				!user.dataValues.status ? res.sendStatus(503) : console.log(user.dataValues.status)
+				// check cache to see if there is a user on the machine
+				var userWaiting = cache.getMachine(headerObj.sid, timer);
+				if (userWaiting === null && headerObj['station-state'] === 'disabled') {
+					//there is no user in cache and the machine is 'disabled'
+					//put the admin on the machine	
+					// call station module HERE //to update station with current user and change to enabled
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'enabled',
+						'badge':  headerObj.badge
+					});
+					res.sendStatus(200);
+				} else if (headerObj['station-state'] === 'enabled') {
+					//assumption is the admin is logging of from the machine
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'disabled'
+					});
+					res.sendStatus(200);
+				} else {
+					// grant privilages to the user	
+					db.grantPrivileges(userWaiting, headerObj.sid).then(function(result) {
+						cache.delete(headerObj.sid);
+						//call station module HERE //to update station with current user and change to enabled
+						res.set("Connection", "close");
+						!result ? res.sendStatus(403) : res.set({
+																							'Content-Type': 'text/plain',
+																							'Station-state': 'enabled', 
+																							'badge': userWaiting
+																						});
+						// remove user from module
+						res.sendStatus(200);
+					});
+				};
+			} else if (user.dataValues.status === 'User' || user.dataValues.status === 'Manager') {
+				// handle user
+				// Assumption is the user is logging off the machine
+				//this needs to check if a user is on the machine
+				if (headerObj['station-state'] === 'enabled') {
+					// call station module HERE //to update station disabled and remove user
+					res.set({
+						'Content-Type': 'text/plain',
+						'Station-State': 'disabled'
+					});
+					res.sendStatus(200);
+				} else {
+					db.getPrivileges(headerObj.badge).then(function(obj) {
+						if (obj) {
+							for (let element of obj) {
+								if (element.sId === headerObj.sid) {
+									res.set({
+										'Content-Type': 'text/plain',
+										'Station-State': 'enabled',
+										'badge': headerObj.badge
+									});
+									return res.sendStatus(200);
+									// call station module HERE //to update station with current user and change to enabled
+								}; 	
+							};
+							// user does not have access to this machine put them in cache
+							let check = cache.save(headerObj.sid, headerObj.badge);
+							res.sendStatus(200);
+						} else {
+							// user does not have access to any machines put them in cache
+							let check = cache.save(headerObj.sid, headerObj.badge);
+							res.sendStatus(200);
```

Comment:
This should be a 403

Reply:
@jstar1 What I remember talking about regarding changing the API spec was that we're now expecting the station to provide it's current state in a `station-state` header in the request. The spec is updated to reflect that. The badge number that identifies the badging user should arrive as the body of the request.

Reply:
@jstar1 @bsm5 regarding manager access, the way I remember Chris explaining it is that managers have the ability to train people on the machines that they're trained on.

Reply:
@mongolsamurai In the requirements doc Managers have the ability to train users and other managers on stations that they themselves have also been trained on.  They also have the ability to grant privileges to the users and other managers after they have trained them.

From what I remember a Manager is just like an Admin with the caveat that not all Managers are trained on all machines and that a Manager should never have to access the admin login.